src/databricks/labs/blueprint/parallel.py

```diff
@@ -130,6 +130,24 @@ def _progress_report(self, _):
                 msg = f"{self._name} {self._completed_cnt}/{total_cnt}, rps: {rps:.3f}/sec"
                 logger.info(msg)
 
+    @staticmethod
+    def _get_result_function_signature(func, name):
+        if isinstance(func, functools.partial):
```

Comment:
```suggestion
        if not isinstance(func, functools.partial):
            return name
```

then you have nesting even less